src/App.tsx

```diff
@@ -15,47 +15,130 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type SortCallback = (a: string, b: string) => number;
+
+enum SortType {
+  LENGTH = 'length',
+  ALPHABETICAL = 'alphabetical',
+  NONE = '',
+}
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>(goodsFromServer);
+  const [sortType, setSortType] = useState<SortType>(SortType.NONE);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const sort = (array: string[], callback: SortCallback) => {
+    const sorted = array.sort(callback);
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+    isReversed
+      ? setGoods(sorted.reverse())
+      : setGoods(sorted);
+  };
+
+  const sortAlphabetically = () => {
+    sort([...goods], (a: string, b: string) => {
+      if (a < b) {
+        return -1;
+      }
+
+      if (a > b) {
+        return 1;
+      }
+
+      return 0;
+    });
+
+    setSortType(SortType.ALPHABETICAL);
+  };
+
+  const sortByLength = () => {
+    let copyGoods = [...goods];
```

Comment:
Inside of this compoment, you just need like 3 handler functions, for changing sortType, changing isReverse and reseting the state.

Do all the sorting in a function outside of the component. The function should accept goods from server, sortType and isReverse. Then you will be able to sort items basing on that (ideally using a switch statement).
Return the goods in proper order and render them inside of jsx